### DISCUSSION TITLE
heroes of might and magic 2 - campaign - the price of loyalty - mission 8

### DISCUSSION THREAD

**Discussion by TLCD17** [ihhub/fheroes2#8764] [General]
I managed to complete the entire first Roland campaign. It wasn't as easy as in the movie, e.g. speedrun. In mission 7, I had to gather a slightly larger army than the speedrunner in the video, so that I could barely defeat the final battle with this army in mission 10. It wasn't difficult for me because of my skills, but because of the systematics of fH2. The speedruner had an easier final fight in this mission, and I was not weaker than him in this mission, and the computer had a larger army in my case than in the speedrunner's.
I have read that in recent days some players have been complaining about the 8th mission of the campaign "The price of loyalty". I've come to this mission myself and I can't complete it. The bonus I got from the "Betrayal's End" mission is "Battle Garb of Anduran" + 5 starting mages. I attempted the 8th mission several times with different results, but the enemy always had a larger number of troops after a few months. The furthest I got was with over 20 titans, and the opponent who sailed to the island had 10 of them and 40 phoenixes. This is not the only strong opponent that is difficult to defeat. Actually, there are a couple of them - usually two or three. In this mission, I tried to take over enemy castles as quickly as possible and recruit titans, possibly in the first weeks, without neglecting the acquisition of raw materials, which were not enough for me to be able to match the computer, which quickly took over my castles, arriving on the islands a few days after capturing the castle. I managed to get 5 locks, but the computer quickly took over my locks and I was left with one lock - the starter one. I had almost no losses in my troops, I had a good army, but the computer was still stronger. Moreover, the yellow computer did not attack the purple one or vice versa, and therefore these few locks of the computer at the beginning are not accessible for me to take over, including their heroes, because the computer recovers them after a few days because of armies that I am unable to defeat especially all not because of the armies in the castles I captured, but because I lack the raw materials to buy the armies in the castles I captured. Even if you try to convert the raw materials into gold and everything into titans after conquering 5 castles and markets, it is still not enough to defeat enemy opponents.
I'm not a beginner in HOMM. I have experience playing this series of games. However, I state that this mission is not possible for me to complete. It's possible that playing on Normal difficulty will make it easier to complete this mission - surprisingly some people can do it on Hard difficulty, but I'm unable to complete this mission on easy difficulty.

**Comment by Alucard648** [comment]
You are still lucky. On my attempts, I got DD`ed on week 3. And if enemy hero attacks you - "Armageddon does 3000 damage"...  And it is curtains.

**Comment by ihhub** [comment]
This is not an issue. Converting it into a discussion.

**Comment by TLCD17** [comment]
I don't know if this is supposed to be a discussion or not a problem, specifically completing the mission of the last campaign "The Price of Loyalty" fH2. In any case, I will ask whether in the 8th mission of the Price of Loyalty campaign the hero can have a larger starting army than currently, or whether work will be carried out to make the opponents play worse in this mission?
I believe this is necessary to complete this mission.

**Reply by Olegarh1a** [reply]
I also can’t complete the mission, I kill 2-3 strong heroes, the computer comes with an even larger army. the developer replies that we simply don’t know how to play on the normal one.
It seems to me that AI has too strong bonuses

**Reply by fheroes2bugs** [reply]
> I also can’t complete the mission, I kill 2-3 strong heroes, the computer comes with an even larger army. the developer replies that we simply don’t know how to play on the normal one. It seems to me that AI has too strong bonuses

@Olegarh1a 8 here is someone doing this map in fheroes: https://www.youtube.com/watch?v=83mwVae2y6g
don't know what version he is playing, but video is only 1 week old

**Reply by minuspinus** [reply]
I passed this mission, I had to choose the help of dragons and nagene the map to have dragons nearby, they joined and I ran to the boss)

**Comment by modo-lv** [comment]
I am also currently struggling with the mission. After reading older discussions on the topic I know that the AI won't summon neutral boats, however that doesn't seem to change much for this map. The green player starts with two shipyards and some boats just reachable from the shore.

Without AI boat summoning the map has clear progression and goals:
1. Defeat the green player before they can grow in strength, take the towns, and with them, access to boats.
2. Use the taken castles to reinforce your armies to defend against yellow/purple trying to recapture the shipyard, to keep them contained to their land mass and eventually push back and take over.
3. Use the full resources of your empire to build an army capable of going up against red.

But now as soon as green unlocks the stone lith, yellow and purple are free to roam the whole map, as they can summon any boat ever used. So instead of the intended challenge of how to contain a stronger force with a weaker one while it grows, it becomes a tedious map-wide cat-and-mouse game with cheese tactics like suicide-bomber heroes. I've played through both original campaigns and all of PoL up to this point without using any such exploits (even PoL 7 I managed to do without blocking any stone liths!), but this map just seems to be impossible to play as originally intended.

**Reply by zenseii** [reply]
I'd like to add that "as originally intended" for this map is a bit hard to judge. "Originally intended" the AI should have a higher chance to learn Dimension Door, and have the map visible from early on. In other words, this map very often had the red player jumping around with its heroes chasing you with hundreds of bone dragons. It's been discussed whether this was a design oversight, similar to how people debate this about purple player getting dimension door in Broken Alliance, or if it was intentional to leave such things up to RNG.

**Reply by modo-lv** [reply]
Of course, there's no way to know for sure what the developers intended, but I think the map's design in this case is very distinct and obvious.

As for the red opponent getting DD in the original game, that has to be an oversight. HoMM's limited randomness adds replay variability and some playthroughs can be made or broken by what spells players get (especially on harder difficulties), but randomly making an entire scenario virtually unwinnable before it even starts? That cannot possibly be intentional design in a game of strategy. Maybe if it was a small standalone map it could be considered a practical joke, but in the final scenario of the main campaign of the official expansion? No way.

**Reply by oleg-derevenetz** [reply]
> But now as soon as green unlocks the stone lith, yellow and purple are free to roam the whole map, as they can summon any boat ever used.

That's not exactly how it works. Human players can summon neutral boats and the boats that were last used by their own hero. AI players can only summon the boats that were last used by their own hero. For instance, if a boat has been used by a green hero, the blue hero will not be able to summon it. Similarly, if a boat has been used by a green hero, and then it has been used by a blue hero (without summoning it, naturally), then the blue hero will be able to summon it later, but the green hero will not be able to summon it anymore.

**Reply by modo-lv** [reply]
OK, I did not know that. That should make it potentially possible to deny boats to yellow and purple, if green doesn't unlock the liths too early. Since I'm assuming they also can't "tell" green to build a boat just for them, the green will only build boats when it needs them for its own heroes.

**Reply by oleg-derevenetz** [reply]
> Since I'm assuming they also can't "tell" green to build a boat just for them, the green will only build boats when it needs them for its own heroes.

A boat built by a kingdom (a castle from a kingdom) immediately belongs to that kingdom, even if no hero has boarded it yet. Only the boats placed on the map by its creator in the map editor are neutral. Therefore yellow and purple cannot summon boats built by green, they can only capture these boats "manually" by finding one of these empty boats on a shore and boarding them.
